File: APPS/reloj/RellotgeJAIS/app/src/main/java/com/example/rellotgejais/IncidenciaAdapter.java
package com.example.rellotgejais;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

public class IncidenciaAdapter extends RecyclerView.Adapter<IncidenciaAdapter.ViewHolder> {

    private List<Incidencia> incidencias;

    public IncidenciaAdapter(List<Incidencia> incidencias) {
        this.incidencias = incidencias;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_incidencia, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Incidencia incidencia = incidencias.get(position);
        holder.tvTiempo.setText(String.format("%02d:%02d", incidencia.getMins(), incidencia.getSeconds()));
        holder.tvNumero.setText(String.valueOf(incidencia.getJerseyNumber()));
        holder.ivIncidencia.setImageResource(incidencia.getImagenResId());
    }

    @Override
    public int getItemCount() {
        return incidencias.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvTiempo, tvNumero;
        ImageView ivIncidencia;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvTiempo = itemView.findViewById(R.id.tvTiempo);
            tvNumero = itemView.findViewById(R.id.inccidenciplayerNum);
            ivIncidencia = itemView.findViewById(R.id.incidenciaImage);
        }
    }
}

File: APPS/reloj/RellotgeJAIS/app/src/main/java/com/example/rellotgejais/screens/IncidenciaAdapter.kt
package com.example.rellotgejais.screens

import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import com.example.rellotgejais.models.PopulatedIncident

class IncidenciaAdapter/*(private val incidencias: List<PopulatedIncident>) */: AppCompatActivity() {
  /*
    RecyclerView.Adapter<IncidenciaAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_incidencia, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val incidencia = incidencias[position]
        holder.tvTiempo.text = String.format("%02d\'", incidencia.raw.mins)
        holder.tvNumero.text = incidencia.player?.dorsal?.toString() ?: "-"
        holder.ivIncidencia.setImageResource(incidencia.imagenResId)
    }

    override fun getItemCount(): Int {
        return incidencias.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvTiempo: TextView = itemView.findViewById(R.id.tvTiempo)
        var tvNumero: TextView = itemView.findViewById<TextView>(R.id.dorsal)
        var ivIncidencia: ImageView = itemView.findViewById(R.id.incidenciaImage)
    }


}


class IncidenciaAdapter(private val incidencias: List<PopulatedIncident>) :
    RecyclerView.Adapter<IncidenciaAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        // val view = LayoutInflater.from(parent.context)
        //   .inflate(R.layout.item_incidencia, parent, false)
        // return ViewHolder(view)
        return ViewHolder(View(parent.context))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        // val incidencia = incidencias[position]
        // holder.tvTiempo.text = String.format("%02d\'", incidencia.raw.mins)
        // holder.tvNumero.text = incidencia.player?.dorsal?.toString() ?: "-"
        // holder.ivIncidencia.setImageResource(incidencia.imagenResId)
    }

    override fun getItemCount(): Int {
        // return incidencias.size
        return 0
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        // var tvTiempo: TextView = itemView.findViewById(R.id.tvTiempo)
        // var tvNumero: TextView = itemView.findViewById(R.id.dorsal)
        // var ivIncidencia: ImageView = itemView.findViewById(R.id.incidenciaImage)
    }
    */
}